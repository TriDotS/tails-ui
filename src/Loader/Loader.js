import styled, { keyframes } from 'styled-components'

import PropTypes from 'prop-types'
import React from 'react'

const spinAnimation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`

const StyledLoader = styled.div`
  animation: ${spinAnimation} 1s linear infinite;
`

const Loader = ({ color }) => (
  <StyledLoader
    className={`border-6 border-${color} bg-grey-light rounded-full w-8 h-8 border-t-4 shadow`}
  />
)

Loader.propTypes = {
  color: PropTypes.string
}

Loader.defaultProps = {
  color: 'teal'
}

export default Loader
