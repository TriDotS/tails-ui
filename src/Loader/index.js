import styled, { keyframes } from 'styled-components'

import React from 'react'

const spin = keyframes`
  0% { 
    transform: rotate(0deg); 
  }
  100% { 
    transform: rotate(360deg); 
  }
`

const LoaderContainer = styled.div`
  border: 5px solid #f3f3f3;
  border-radius: 50%;
  border-top: 5px solid #3498db;
  width: 30px;
  height: 30px;
  animation: ${spin} 1s linear infinite;
`

const Loader = () => (
  <LoaderContainer />
)

export default Loader