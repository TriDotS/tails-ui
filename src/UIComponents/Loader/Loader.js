import styled, { keyframes } from 'styled-components'

import PropTypes from 'prop-types'
import React from 'react'

const spinAnimation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`

const StyledLoader = styled.div`
  animation: ${spinAnimation} 1s linear infinite;
`

const Loader = ({ primary, secondary, success, info, danger }) => {
  let loaderColor
  if (primary) {
    loaderColor = 'border-teal'
  } else if (secondary) {
    loaderColor = 'border-pink-light'
  } else if (success) {
    loaderColor = 'border-green-light'
  } else if (info) {
    loaderColor = 'border-blue-light'
  } else if (danger) {
    loaderColor = 'border-red-lighter'
  } else {
    loaderColor = 'border-grey'
  }
  return (
    <StyledLoader
      className={`border-6 ${
        loaderColor
      } bg-grey-light rounded-full w-8 h-8 border-t-4 shadow`}
    />
  )
}

Loader.propTypes = {
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  success: PropTypes.bool,
  info: PropTypes.bool,
  danger: PropTypes.bool
}

export default Loader
