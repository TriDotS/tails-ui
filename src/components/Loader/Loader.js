import PropTypes from 'prop-types'
import React from 'react'
import { spinAnimation } from '../../utils/animations'
import styled from 'styled-components'

const StyledLoader = styled.div`
  animation: ${spinAnimation} 1s linear infinite;
`

const Loader = ({ primary, secondary, success, info, danger }) => {
  let loaderColor
  if (primary) {
    loaderColor = 'border-blue'
  } else if (secondary) {
    loaderColor = 'border-pink-light'
  } else if (success) {
    loaderColor = 'border-green-light'
  } else if (info) {
    loaderColor = 'border-indigo-light'
  } else if (danger) {
    loaderColor = 'border-red-lighter'
  } else {
    loaderColor = 'border-grey'
  }
  return (
    <StyledLoader
      className={`border-6 ${
        loaderColor
      } bg-grey-light rounded-full w-8 h-8 border-t-4 shadow`}
    />
  )
}

Loader.propTypes = {
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  success: PropTypes.bool,
  info: PropTypes.bool,
  danger: PropTypes.bool
}

export default Loader
