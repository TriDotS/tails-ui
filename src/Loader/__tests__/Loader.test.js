import Loader from '../Loader'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Loader />, div)
})

test('loader snapshot', () => {
  const tree = renderer.create(<Loader />).toJSON()
  expect(tree).toMatchSnapshot()
})
