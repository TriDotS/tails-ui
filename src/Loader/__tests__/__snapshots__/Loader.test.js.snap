// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader snapshot 1`] = `
<div
  className="border-6 border-teal bg-grey-light rounded-full w-8 h-8 border-t-4 shadow sc-bdVaJa ixIJhe"
/>
`;
